### ISSUE TITLE
react/display-name false positive when using React.memo

### ISSUE THREAD

**Issue by Raigen** [yannickcr/eslint-plugin-react#2105]
`eslint-plugin-react@7.12.1`

We have a functional component that is exported with `React.memo`
```jsx
import React from 'react'
import { string } from 'prop-types'

function Component({ world }) {
  return <div>Hello {world}</div>
}

Component.propTypes = {
  world: string,
}

export default React.memo(Component)
```

This triggers the `react/display-name` error after the update from `7.11.1`
```
[...]/Component.jsx
  12:16  error  Component definition is missing display name  react/display-name
```
The exported components `displayName` is `Memo(Component)`.

Without `React.memo` it recognizes the `displayName`.
```jsx
export default Component
```

**Comment by ljharb** [comment]
cc @jomasti

**Comment by samsch** [comment]
The fix in 7.12.3 works for `export default React.memo(Component);`, but still gives a false positive if you wrap the function directly:
```jsx
import React from 'react';

const Test = React.memo(props => (
  <div>{props.children}</div>
));
export default Test;
```

**Comment by ljharb** [comment]
@samsch thanks, could you file that as a new issue?
